### DISCUSSION TITLE
What is the end goal?

### DISCUSSION THREAD

**Discussion by nrdodson** [immich-app/immich#3045] [General]
Hey everyone; I have used and loved Immich now for a little while, and of course the extremely active development has lent itself well to rapid platform growth (and also breaking lots of things, in the good kind of way).

I guess I am just curious if there is a point on the roadmap where "MVP" will be considered met, and there will be a shift in focus towards stability and less so of adding features?

I am super excited to fully switch over to Immich, but of course that is not really possible in its current state.

**Comment by alextran1502** [comment]
Hello, to give you some idea of things we want to achieve, below is a list I have in mind.

- Fully functional facial recognition features, basically completing this list #2472
- Using timeline view performance for all views on the web.
- Optimize performance for the large gallery on the mobile app.
- More file-watching feature to the read-only mechanism (not sure if this is relevant to your use case)


Those are some big-ticket items I can think of. A lot of those items will require some refactoring of the code base

**Reply by nrdodson** [reply]
That all sounds great; looking forward to it :D I will definitely be contributing where I can moving forward.

**Reply by titro** [reply]
Hi Alex,

good to know. So the gallery watcher feature is back to work on it in the near future :-)?

**Reply by alextran1502** [reply]
@titro the initial implementation is already in. You can read about it here https://immich.app/docs/features/read-only-gallery
